addons/cargo/functions/fnc_onMenuOpen.sqf

```diff
@@ -22,37 +22,64 @@ params ["_display"];
-    if (isnil "_display") exitWith {
-        [_this select 1] call CBA_fnc_removePerFrameHandler;
+
+    if (isNil "_display") exitWith {
+        _pfhID call CBA_fnc_removePerFrameHandler;
     };
 
-    if (isNull GVAR(interactionVehicle) || {(([ACE_player, GVAR(interactionVehicle)] call EFUNC(interaction,getInteractionDistance)) >= MAX_LOAD_DISTANCE) && {(vehicle ACE_player) != GVAR(interactionVehicle)}}) exitWith {
+    // Close menu if in invalid state
+    if (
+        !alive _vehicle ||
+        {locked _vehicle >= 2} ||
+        {!(_vehicle getVariable [QGVAR(hasCargo), getNumber (configOf _vehicle >> QGVAR(hasCargo)) == 1])} ||
+        {(([ACE_player, _vehicle] call EFUNC(interaction,getInteractionDistance)) >= MAX_LOAD_DISTANCE) && {(vehicle ACE_player) != _vehicle}}
+    ) exitWith {
         closeDialog 0;
-        [_this select 1] call CBA_fnc_removePerFrameHandler;
+
+        _pfhID call CBA_fnc_removePerFrameHandler;
     };
 
-    private _loaded = GVAR(interactionVehicle) getVariable [QGVAR(loaded), []];
     private _ctrl = _display displayCtrl 100;
     private _label = _display displayCtrl 2;
 
+    // Remove previous entries
     lbClear _ctrl;
+
+    // Display item names
+    private _displayName = "";
+    private _itemSize = 0;
+    private _index = -1;
+
     {
-        private _class = if (_x isEqualType "") then {_x} else {typeOf _x};
-        private _displayName = [_x, true] call FUNC(getNameItem);
-        if (GVAR(interactionParadrop)) then {
-            _ctrl lbAdd format ["%1 (%2s)", _displayName, GVAR(paradropTimeCoefficent) * ([_class] call FUNC(getSizeItem))];
+        _displayName = [_x, true] call FUNC(getNameItem);
+        _itemSize = _x call FUNC(getSizeItem);
+
+        if (_itemSize >= 0) then {
+            _index = if (GVAR(interactionParadrop)) then {
+                _ctrl lbAdd format ["%1 (%2s)", _displayName, GVAR(paradropTimeCoefficent) * _itemSize]
+            } else {
+                _ctrl lbAdd _displayName
+            };
+
+            _ctrl lbSetTooltip [_index, format [LLSTRING(sizeMenu), _itemSize]];
         } else {
-            _ctrl lbAdd _displayName;
-        };
+            // If item has a size < 0, it means it's not loadable
+            _index = _ctrl lbAdd _displayName;
 
-        true
-    } count _loaded;
+            _ctrl lbSetTooltip [_index, LLSTRING(unloadingImpossible)];
+            _ctrl lbSetColor [_index, [1, 0, 0, 1]]; // set text to red
+            _ctrl lbSetSelectColor [_index, [1, 0, 0, 1]];
+        };
+    } forEach (_vehicle getVariable [QGVAR(loaded), []]);
 
-    _label ctrlSetText format[localize LSTRING(labelSpace), [GVAR(interactionVehicle)] call DFUNC(getCargoSpaceLeft)];
-}, 0, []] call CBA_fnc_addPerFrameHandler;
+    _label ctrlSetText format [LLSTRING(labelSpace), (_vehicle call FUNC(getCargoSpaceLeft)) max 0];
+}, 0.25, GVAR(interactionVehicle)] call CBA_fnc_addPerFrameHandler;
```

Comment:
Maybe move PFH to separate function?

Reply:
Also, don't like the increased delay. Could get iffy with multiple people unloading from same vehicle.

Reply:
> Maybe move PFH to separate function?

It feels unnecessary, as the nearly the entire function is the definition of the PFH.

> Also, don't like the increased delay. Could get iffy with multiple people unloading from same vehicle.

Fair, I just wanted to avoid looking up info, such as size and cargo space, every frame.

Reply:
Could be solved by updating menu on unload/load global event instead but it's more refactoring

Reply:
I don't think it's worth it, at least for now.